datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -109,6 +109,11 @@
   protected Consumer<?, ?> _consumer;
   protected final Set<TopicPartition> _consumerAssignment = new HashSet<>();
 
+  // TopicPartitions which have seen exceptions on send. Access to this map must be synchronized.
+  // A ConcurrentHashMap is not used here due to the need for having more than one operation performed together as an
```

Comment:
Do we need this extra explanation? It might become a norm every time to put this in comment. I understand this will make the understanding easier, but do we then want to put every time going forward. It might become cumbersome.

Reply:
Deepthi pointed out that in the future someone may accidentally change this to a concurrent hashmap, and having a comment explaining why may help prevent that. I think it’s okay to explain when important so that people aren’t left wondering why one data structure is chosen over another. What do you think?

Reply:
Sure. sounds good.